### PR TITLE
1b- Create user preferences form - Milad

### PR DESCRIPTION
**Pull request by Miladepour**
## Description

Create user story preferences and link to the use form.

## Related to

Feature ticket 3 / [(1-b) User preference page #3#7](https://github.com/Miladepour/SyntaxSquad-Project-Kunji/issues/7)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Review by KptCnyt — APPROVED** [review]

File: client/src/pages/user-preferences/UserPreferences.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+							</option>
+						);
+					})}
+				</Form.Select>
+                <Form.Text className="text-muted">
+                {errors.location && errors.location.message}
+                </Form.Text>
+			</Form.Group>
+			<div className="container-btn">
+				<Link className="btn btn-primary" to="#" onClick={handleSubmit(onSubmit)}>
```
Comment (EneVladMihai) [inline client/src/pages/user-preferences/UserPreferences.js:72]: 1.This is a button that will handle form submission (we don't technically submit anything at this point but this button will take the form inputs and give us a result). If we use `<button type'submit'/>` the call to handleSubmit will be automatically handled. [see docs example](https://react-bootstrap.netlify.app/forms/overview/#overview).

**Review by EneVladMihai — CHANGES_REQUESTED** [review]

File: client/src/pages/user-preferences/UserPreferences.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				<Form.Select
+					aria-label="Default select example"
+					className="w-50"
+					type="text"
+					name="service"
+					{...register("service", {
+						required: "Please select your service",
+					})}
+                    placeholder="Select here..."
+                    onChange={(e) => console.log(e.target.value)}
```
[the remark this answers is not in the document]
Comment (Miladepour) [reply client/src/pages/user-preferences/UserPreferences.js]: The link was very helpful thanks.
@haroon-ali-dev @baharbazargan62 @KptCnyt

File: client/src/pages/user-preferences/UserPreferences.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    <option value="">Select here...</option>
+					{Data.services.map((service) => {
+						return (
+							<option key={service} value={service}>
+								{service}
+							</option>
+						);
+					})}
+				</Form.Select>
+                <Form.Text className="text-muted">
```
[the remark this answers is not in the document]
Comment (Miladepour) [reply client/src/pages/user-preferences/UserPreferences.js]: I am considering changing the color to red at the moment, but I plan to add another method in the next iteration.
Thanks.

File: client/src/pages/user-preferences/UserPreferences.js (hunk 1 above)
Comment (Miladepour) [reply client/src/pages/user-preferences/UserPreferences.js]: Thanks for your suggestion.
I've changed it.

File: client/src/components/UserForm.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { useState } from "react";
 import { Link } from "react-router-dom";
-import Form from 'react-bootstrap/Form';
+import Form from "react-bootstrap/Form";
```
[the remark this answers is not in the document]
Comment (Miladepour) [reply client/src/components/UserForm.js:3]: This is based on .prettierrc file in root:
{
	"arrowParens": "always",
	"bracketSpacing": true,
	"endOfLine": "lf",
	"semi": true,
	"singleQuote": false,
	"trailingComma": "es5",
	"useTabs": true
}

File: client/src/pages/user-preferences/UserPreferences.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+							</option>
+						);
+					})}
+				</Form.Select>
+				<Form.Text className="error-message">
+					{errors.location && errors.location.message}
+				</Form.Text>
+			</Form.Group>
+			<div className="container-btn">
+				<Button variant="btn btn-primary" type="submit">
```
Comment (EneVladMihai) [inline client/src/pages/user-preferences/UserPreferences.js:66]: I think you don't need the `variant`  prop here. The value `btn btn-primary` is proably for `className`. Now that you're using a Button you already have these 2 classes by default.

File: client/src/components/UserForm.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (EneVladMihai) [reply client/src/components/UserForm.js:3]: Ok, this makes sense. I'm wondering if it runs automatically for everyone when they commit, but let's come back to it later. This shouldn't block the PR from getting approved.

File: client/src/pages/user-preferences/UserPreferences.js (hunk 5 above)
Comment (Miladepour) [reply client/src/pages/user-preferences/UserPreferences.js:66]: I don't know where this coming from 😠 , I fix that now.

File: client/src/pages/user-preferences/UserPreferences.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+					<option value="">Select here...</option>
+					{Data.services.map((service) => {
+						return (
+							<option key={service} value={service}>
+								{service}
+							</option>
+						);
+					})}
+				</Form.Select>
+				<Form.Text className="error-message">
```
Comment (EneVladMihai) [inline client/src/pages/user-preferences/UserPreferences.js:37]: There is some built-in functionality in Boostrap to achieve this.
Here is an example usage https://github.com/Miladepour/SyntaxSquad-Project-Kunji/pull/16/files#diff-d2e54251fe00b6c1aa216984652dbe359d580b1379364133310be76e0250289bR55

The logic to show it is contolled via the `isInvalid` from the `<Form.Control>`

Comment (Miladepour) [reply client/src/pages/user-preferences/UserPreferences.js]: Solved.

**Review by EneVladMihai — APPROVED** [review]

**Pull request merged**